src/Microsoft.TemplateEngine.Edge/Template/MatchKind.cs

```diff
@@ -1,15 +1,56 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+
+using System;
+
 namespace Microsoft.TemplateEngine.Edge.Template
 {
-
+    /// <summary>
+    /// Represents match status.
+    /// </summary>
     public enum MatchKind
```

Comment:
I'm wondering if we should also deprecate this enum and make it string, as `MatchLocation`. 
The reason is that then host will be able to define their own values for filters.